arcgis-ios-sdk-samples/Layers/Group layers/LayersTableViewController.swift

```diff
@@ -87,46 +144,38 @@ class LayersTableViewController: UITableViewController, GroupLayersCellDelegate,
     func didToggleSwitch(_ sectionView: GroupLayersSectionView, isOn: Bool) {
         guard let section = tableView.section(forHeaderView: sectionView) else { return }
         layers[section].isVisible = isOn
-        updateSwitchForRows(in: section, isEnabled: isOn)
+        let rowsInSection = tableView.numberOfRows(inSection: section)
+        let rowArray = 0...rowsInSection - 1
+        let indexPaths = rowArray.map { IndexPath(row: $0, section: section) }
+        reloadRows(at: indexPaths)
```

Comment:
I think the table view's method to reload a section is more efficient than reloading all the rows in the section. It would probably be better to add a custom function to reload a section, just like you did for reloading rows. Could you please do that?

Reply:
Both of these functions are only used once -- would it be better to just remove the observer right before calling `reloadRows` in the original function rather than make two separate functions?

Reply:
I was under the impression that they were called in multiple places. Since they're not, then I suppose the custom functions aren't needed.